### PR TITLE
Management command to update voucher names and make them unique

### PR DESCRIPTION
**Pull request by zubair-ce07**
# ⛔️ MAIN BRANCH WARNING! 2U EMPLOYEES must make branches against the 2u/main BRANCH

- [x] I have checked the branch to which I would like to merge.

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

**Review by christopappas — CHANGES_REQUESTED** [review]
Please add a test that runs the management command twice to prove it is idempotent. 

Additionally, should this be aware of the name length much like your database migration is? I think that needs to be reflected in this code, and a test should be added for that as well.

**Review by christopappas — APPROVED** [review]
thank you for adding the additional tests

File: ecommerce/extensions/voucher/management/commands/tests/test_update_voucher_names.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        """
+        call_command('update_voucher_names')
+
+        vouchers = Voucher.objects.all()
+        assert vouchers.count() == 3
+
+        for voucher in vouchers:
+            assert voucher.name == f'{voucher.id} - {self.voucher_name}'
+
+    def test_update_voucher_names_long_name(self):
```
Comment (julianajlk) [inline ecommerce/extensions/voucher/management/commands/tests/test_update_voucher_names.py:48]: Nice addition!

**Review by julianajlk — APPROVED** [review]
LGTM

**Pull request merged**